llvm/lib/Target/AIE/AIEPostPipeliner.cpp

```diff
@@ -160,8 +178,10 @@ void PostPipeliner::scheduleNode(SUnit &SU, int Cycle) {
 // Check resources. We only insert at the position modulo II. Since we insert
 // all iterations separately, the resources that wrap around accumulate in the
 // overflow area, causing conflicts when inserting future iterations
-int PostPipeliner::fit(MachineInstr *MI, int Earliest, int NTries, int II) {
-  for (int C = Earliest; C < Earliest + NTries; C++) {
+int PostPipeliner::fit(MachineInstr *MI, int First, int Last, int II) {
+  const int Step = First > Last ? -1 : 1;
+  LLVM_DEBUG(dbgs() << "   " << First << ", " << Last << ", " << Step << "\n");
+  for (int C = First; C != Last; C += Step) {
```

Comment:
Maybe have `C < Last`? Or do we have a guarantee that `Step` divides `Last-First`?

Reply:
Oh boy, I thought that `Step` came in as an argument, but it's just defined one line above 🙈